code/_globalvars/lists/ai_laws.dm

```diff
@@ -0,0 +1,48 @@
+GLOBAL_LIST_INIT(lawset_laws, get_lawset_laws())
+
+/proc/get_lawset_laws()
+	. = list()
+
+	for(var/lawset in subtypesof(/datum/ai_laws))
+		var/datum/ai_laws/lawset_datum = new lawset
+
+		.[lawset] = list()
+
+		.[lawset][LAW_NAME] = lawset_datum.name
+
+		.[lawset][LAW_ZEROTH] = lawset_datum.zeroth
+
+		.[lawset][LAW_HACKED] = list()
+
+		.[lawset][LAW_HACKED].Add(lawset_datum.hacked)
```

Comment:
`Add` works here, but I can change the syntax if requested.

Reply:
Gonna do some testing locally, something is wrong here.

Reply:
The linter doesn't like the syntax but it works.
`warning: proc call requires static type: "Add"`

Reply:
try `.[lawset][LAW_HACKED] = lawset_datum.hacked.Copy()` instead of this store-then-`.Add` dance

Reply:
although I'm not sure why this is an assoc list anyways instead of just storing instances `/datum/ai_laws`